lib/rules/use-isnan.js

```diff
@@ -21,7 +21,10 @@ const astUtils = require("./utils/ast-utils");
  * @returns {boolean} `true` if the node is 'NaN' identifier.
  */
 function isNaNIdentifier(node) {
-    return Boolean(node) && node.type === "Identifier" && node.name === "NaN";
+    return Boolean(node) && (node.type === "Identifier" && node.name === "NaN" ||
+        (node.type === "MemberExpression" &&
+        node.object.type === "Identifier" && node.object.name === "Number" &&
+        node.property.type === "Identifier" && node.property.name === "NaN"));
```

Comment:
```suggestion
    return Boolean(node) && (
        astUtils.isSpecificId(node, "NaN") ||
        astUtils.isSpecificMemberAccess(node, "Number", "NaN")
    );
```

False positive (we should account for `MemberExpression#computed`):

```js
/* eslint use-isnan: error */

x === Number[NaN]; // false positive
```

False negative (we should check `ChainExpression#expression`):

```js
/* eslint use-isnan: error */

x === Number?.NaN; // false negative

```

This could be seen as a false negative:

```js
/* eslint use-isnan: error */

x === Number["NaN"]; // false negative
```

Reply:
Updated code and added corresponding test cases 👍🏻